src/LocalField/Eisenstein.jl

```diff
@@ -0,0 +1,209 @@
+end
+
+function NumberField(f::fmpz_poly, s::AbstractString; cached::Bool = true, check::Bool = true)
+  Qx, x = PolynomialRing(FlintQQ, string(parent(f).S))
+  return NumberField(Qx(f), s, cached = cached, check = check)
+end
+
+function NumberField(f::fmpz_poly; cached::Bool = true, check::Bool = true)
+  Qx, x = PolynomialRing(FlintQQ, string(parent(f).S))
+  return NumberField(Qx(f), cached = cached, check = check)
+end
+
+@doc Markdown.doc"""
+    radical_extension(n::Int, gen::Integer; cached::Bool = true, check::Bool = true) -> EisensteinField, nf_elem
+    radical_extension(n::Int, gen::fmpz; cached::Bool = true, check::Bool = true) -> EisensteinField, nf_elem
+
+The number field with defining polynomial $x^n-gen$.
+"""
+function radical_extension(n::Int, gen::Integer; cached::Bool = true, check::Bool = true)
+  return radical_extension(n, fmpz(gen), cached = cached, check = check)
+end
+
+function radical_extension(n::Int, gen::fmpz; cached::Bool = true, check::Bool = true)
+  kx, x = FlintQQ["x"]
+  return number_field(x^n - gen, cached = cached, check = check)
+end
+
+@doc Markdown.doc"""
+    cyclotomic_field(n::Int) -> EisensteinField, nf_elem
+
+The $n$-th cyclotomic field defined by the $n$-the cyclotomic polynomial.
+"""
+function cyclotomic_field(n::Int; cached::Bool = true)
+  return CyclotomicField(n, "z_$n", cached = cached)
+end
+
+# TODO: Some sort of reference?
+@doc Markdown.doc"""
+    wildanger_field(n::Int, B::fmpz) -> EisensteinField, nf_elem
+
+Returns the field with defining polynomial $x^n + \sum_{i=0}^{n-1} (-1)^{n-i}Bx^i$.
+These fields tend to have non-trivial class groups.
+"""
+function wildanger_field(n::Int, B::fmpz; cached::Bool = true)
+  Qx, x = PolynomialRing(FlintQQ, "x", cached = false)
+  f = x^n
+  for i=0:n-1
+    f += (-1)^(n-i)*B*x^i
+  end
+  return NumberField(f, "_\$", cached = cached)
+end
+
+function wildanger_field(n::Int, B::Integer; cached::Bool = true)
+  return wildanger_field(n, fmpz(B), cached = cached)
+end
+
+@doc Markdown.doc"""
+    quadratic_field(d::Integer) -> EisensteinField, nf_elem
+    quadratic_field(d::fmpz) -> EisensteinField, nf_elem
+
+Returns the field with defining polynomial $x^n -d$.
+"""
+function quadratic_field(d::fmpz; cached::Bool = true, check::Bool = true)
+  Qx, x = PolynomialRing(FlintQQ)
+  if nbits(d) > 100
+    a = div(d, fmpz(10)^(ndigits(d, 10) - 4))
+    b = mod(abs(d), 10^4)
+    s = "sqrt($a..($(nbits(d)) bits)..$b)"
+  else
+    s = "sqrt($d)"
+  end
+  q, a = number_field(x^2-d, s, cached = cached, check = check)
+  set_special(q, :show => show_quad)
+  return q, a
+end
+
+function show_quad(io::IO, q::EisensteinField)
+  d = trail(q.pol)
+  if d > 0
+    print(io, "Real quadratic field by ", q.pol)
+  else
+    print(io, "Imaginary quadratic field by ", q.pol)
+  end
+end
+
+function quadratic_field(d::Integer; cached::Bool = true, check::Bool = true)
+  return quadratic_field(fmpz(d), cached = cached, check = check)
+end
```

Comment:
Why did these functions move here?

Reply:
They didn't actually, the original functions in the NumberField library are still there. The "Eisenstein.jl" file is not included in the module via an "include(...)" in LocalFields.jl. 

Rather, because Eisenstein fields behave similarly to number fields, I adapt and specialize the interface when it makes sense, as well as try to mimic the style to maintain code consistency.